modules/webclient/src/main/scala/scaladex/client/Client.scala

```diff
@@ -84,6 +93,33 @@ object Client {
               }
           }
       }
+
+      val repoRequest: Request = new Request(
+        s"https://api.github.com/repos/$organization/$repository",
+        new RequestInit {
+          headers = headersWithCreds.toJSDictionary
+        }
+      )
+      fetch(repoRequest).toFuture
+        .flatMap { res =>
+          if (res.status == 200) {
+            res.text().toFuture
+          } else {
+            Future.successful("{\"default_branch\": \"master\"}")
+          }
```

Comment:
`
For the fallback, I would avoid creating a JSON string if we then need to parse it and extract one field.

```suggestion
          def extractDefaultBranch(text: String): String =
            js.JSON.parse(text).asInstanceOf[Repo].default_branch
          if (res.status == 200) res.text().map(extractDefaultBranch).toFuture
          else Future.successful("master")
```